src/Umbraco.Infrastructure/Cache/DefaultRepositoryCachePolicy.cs

```diff
@@ -137,7 +139,7 @@ public override void Delete(TEntity entity, Action<TEntity> persistDeleted)
 
         TEntity? entity = performGet(id);
```

Comment:
I might be doing something incorrect here, but during my tests, this was still being called even after the item was cached, and therefore the DB was still being accessed as well.

After debugging and looking at the code, this seems in fact logic because, if we cache a null value, the variable `fromCache` will be `null` and the check on line 135 will therefore be false. This means that we will not return the value retrieved from cache (line 137) but go on to line 140 (and DB access) like before.

Does that make any sense or did I oversee something 😅 ?